addons/account_edi/models/account_edi_document.py

```diff
@@ -159,10 +159,10 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                         'blocking_level': False,
                     })
 
-                    if move.is_invoice(include_receipts=True) and move.state == 'posted':
+                    if (move.is_invoice(include_receipts=True) or move.payment_id or move.statement_line_id) and move.state == 'posted':
```

Comment:
```suggestion
                    if move.state == 'posted':
``` I will have the case elsewhere anyways and not sure why we try to filter here.  (it should not have any impact) @smetl